### PR TITLE
feat: update algorithm for curated tag list

### LINKED ISSUE #2483: [Server] - Tag recommendation update

### PR DESCRIPTION
**Pull request by Carolusian**
fix #2483

### CODE REVIEW

**Review by tx0c — APPROVED** [review]
Looks good, the PR is small (with only 2 commits 1 file change), in the case, a rebase is better than merge from develop; ( I clicked rebase from the github web), or you may squash it

we can prepare a hotfix for this

File: db/migrations/20220420023434_update_curation_tag_materialized.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+const view = 'curation_tag_materialized'
+
+exports.up = async (knex) => {
+  // drop current materialized view
+  await knex.raw(`DROP MATERIALIZED VIEW IF EXISTS ${view}`)
+
+  // recreate new materialized view
+  await knex.raw(`
+    CREATE MATERIALIZED VIEW ${view} AS
+    SELECT t.id, 1/(SUM(t1.sum_read_time)+1) AS uuid, SUM(t1.sum_read_time) AS sum_read_time_top_n
```
Comment (tx0c) [inline db/migrations/20220420023434_update_curation_tag_materialized.js:14]: `RANDOM() AS uuid` ?

**Review by tx0c — CHANGES_REQUESTED** [review]
I have test-run this, it lost some randonness at every hour `REFRESH MATERIALED VIEW`, may still need to think about it

File: db/migrations/20220420023434_update_curation_tag_materialized.js (hunk 1 above)
Comment (Carolusian) [reply db/migrations/20220420023434_update_curation_tag_materialized.js]: @tx0c you are correct, I added the randomness back. please check.

**Review by jeremyOk — APPROVED** [review]

**Pull request merged**